### PR TITLE
sink(ticdc): refine sink interface and add init method

### LINKED ISSUE #5197: ticdc: refine sink interface add init method
refine cdc sink interface
add an init method to init and clear up old values 
remove drawback from buffer sink

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

We need to clean up the old values of the table checkpoint, otherwise when the table is dispatched back again, it may read the old values.  See: https://github.com/pingcap/tiflow/issues/4464#issuecomment-1085385382. 
To guarantee the value is cleared, we clear old values when the table  pipeline is created.

Issue Number: close #5197 ref https://github.com/pingcap/tiflow/issues/4464

### What is changed and how it works?
refine cdc sink interface
add an init method to init and clear up old values 
remove drawback from buffer sink

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Review by nongfushanquan — COMMENTED** [review]
whether the test case in issue 4464  can be reproduced after this pr ?

File: cdc/sink/manager.go [hunk 1]
```diff
@@ -111,30 +110,13 @@ func (m *Manager) destroyTableSink(ctx context.Context, tableID model.TableID) e
 	m.tableSinksMu.Lock()
 	delete(m.tableSinks, tableID)
 	m.tableSinksMu.Unlock()
-	callback := make(chan struct{})
-	select {
-	case <-ctx.Done():
-		return ctx.Err()
-	case m.drawbackChan <- drawbackMsg{tableID: tableID, callback: callback}:
```
Comment (hi-rustin) [inline cdc/sink/manager.go:118]: After we remove this mechanism, how do we ensure that it will stop?

**Comment by sdojjy** [comment]
> whether the test case in issue 4464 can be reproduced after this pr ?

this pr will fix that issue by cleaning the old checkpoint value.

File: cdc/sink/mysql.go [hunk 2]
```diff
@@ -510,6 +510,30 @@ func (s *mysqlSink) dispatchAndExecTxns(ctx context.Context, txnsGroup map[model
 	s.notifyAndWaitExec(ctx)
 }
 
+func (s *mysqlSink) Init(tableID model.TableID) error {
+	s.cleanTableResource(tableID)
+	return nil
+}
+
+func (s *mysqlSink) cleanTableResource(tableID model.TableID) {
```
Comment (hi-rustin) [inline cdc/sink/mysql.go:518]: Can we add some tests to cover these cleanups to really take effect?

**Review by hi-rustin — APPROVED** [review]
rest LGTM

Approved in advance, but adding more tests is better! Thanks for working on this.

File: cdc/sink/mysql.go (hunk 2 above)
Comment (sdojjy) [reply cdc/sink/mysql.go]: done

**Review by 3AceShowHand — APPROVED** [review]

**Pull request merged**